pika/adapters/blocking_connection.py

```diff
@@ -700,6 +700,15 @@ def channel(self, channel_number=None):
 
         return channel
 
+    def __enter__(self):
+        # Prepare `with` context
+        return self
+
+    def __exit__(self, *args):
+        # Close connection after `with` context
+        self.close()
```

Comment:
If exiting due to SystemExit, KeyboardInterrupt, or other non-Exception-derived exceptions, we most likely don't want to perform anything graceful that slows down the exit of the app. Anyone want to chime in?